planner_cspace/src/planner_3d.cpp

```diff
@@ -441,76 +448,109 @@ class Planner3dNode
+
+      const float range_overshoot = ec_rough_[0] * (range_ + local_range_ + longcut_range_);
 
-      std::vector<Astar::PriorityVec> centers;
-      for (size_t i = 0; i < static_cast<size_t>(num_task_); ++i)
+      while (true)
       {
-        if (open.size() < 1)
+#pragma omp barrier
+#pragma omp single
+        {
+          centers.clear();
+          for (size_t i = 0; i < static_cast<size_t>(num_cost_estim_task_);)
+          {
+            if (open.size() < 1)
+              break;
+            auto center = open.top();
+            open.pop();
+            if (center.p_raw_ > g[center.v_])
+              continue;
+            if (center.p_raw_ - range_overshoot > g[s_rough])
+              continue;
+            centers.push_back(std::move(center));
+            ++i;
+          }
+        }  // omp single
+
+        if (centers.size() == 0)
           break;
-        const auto center = open.top();
-        open.pop();
-        if (center.p_raw_ > g[center.v_])
-          continue;
-        if (center.p_raw_ - ec_rough_[0] * (range_ + local_range_ + longcut_range_) > g[s_rough])
-          continue;
-        centers.push_back(center);
-      }
-#pragma omp parallel
-      {
-        std::list<Astar::GridmapUpdate> updates;
+        updates.clear();
 
 #pragma omp for schedule(static)
         for (auto it = centers.cbegin(); it < centers.cend(); ++it)
         {
           const Astar::Vec p = it->v_;
-          const float c = it->p_raw_;
 
-          for (const Astar::Vec& d : search_diffs)
+          for (const SearchDiffs& ds : search_diffs)
           {
-            Astar::Vec next = p + d;
-            next.cycleUnsigned(map_info_.angle);
+            const Astar::Vec d = ds.d;
+            const Astar::Vec next = p + d;
 
             if (static_cast<size_t>(next[0]) >= static_cast<size_t>(map_info_.width) ||
                 static_cast<size_t>(next[1]) >= static_cast<size_t>(map_info_.height))
               continue;
+
+            float cost = euclidCost(d, ec_rough_);
+
             const float gnext = g[next];
-            if (gnext < 0)
-              continue;
 
-            float cost = 0;
+            if (gnext < g[p] + cost)
+            {
+              // Skip as this search task has no chance to find better way.
+              continue;
+            }
 
             {
               float sum = 0, sum_hist = 0;
-              const float grid_to_len = d.gridToLenFactor();
-              const int dist = d.len();
-              const float dpx = static_cast<float>(d[0]) / dist;
-              const float dpy = static_cast<float>(d[1]) / dist;
-              Astar::Vec pos(static_cast<int>(p[0]), static_cast<int>(p[1]), 0);
-              int i = 0;
-              for (; i < dist; i++)
+              bool collision = false;
+              for (const auto& d : ds.pos)
               {
+                const Astar::Vec pos = p + d;
                 const char c = cm_rough_[pos];
                 if (c > 99)
```

Comment:
What does '99' mean here?

Reply:
It's grid occupancy value in OccupancyGrid ROS message. 100 means 100% occupied.

Reply:
I see, thanks.